### PR TITLE
RD-3115

### PR DESCRIPTION
**Pull request by yuhanniu**
Setters for ultimateDebtor, ultimateCreditor and purposeCode

### CODE REVIEW

File: core/src/main/java/io/token/tokenrequest/TokenRequest.java [hunk 1]
```diff
@@ -382,6 +381,48 @@ public TransferBuilder setChargeBearer(ChargeBearer chargeBearer) {
                     .setChargeBearer(chargeBearer);
             return this;
         }
+
+        /**
+         * Optional. Sets the ultimate party to which an amount of money is due.
```
Comment (gianluca-token) [inline core/src/main/java/io/token/tokenrequest/TokenRequest.java]: "to which the money is due"?

**Comment by zixunjian** [comment]
what does "ultimate debtor/creditor" mean?

File: core/src/main/java/io/token/tokenrequest/TokenRequest.java (hunk 1 above)
Comment (yuhanniu) [reply core/src/main/java/io/token/tokenrequest/TokenRequest.java]: this is the ISO 20022 definition, does it look weird?

**Comment by yuhanniu** [comment]
@zixunjian my understanding is for example if paypal is charging on behalf of a merchant, paypal is the creditor, the merchant is the ultimate creditor.

File: core/src/main/java/io/token/tokenrequest/TokenRequest.java (hunk 1 above)
Comment (yuhanniu) [reply core/src/main/java/io/token/tokenrequest/TokenRequest.java]: changed, i think what you have makes more sense in this context.

**Pull request merged**